socketio/examples/async_transmitter.rs

```diff
@@ -0,0 +1,116 @@
+use futures_util::FutureExt;
+use rust_socketio::{
+    asynchronous::{Client as SocketIOClient, ClientBuilder as SocketIOClientBuilder},
+    Error as SocketIOError, Payload,
+};
+use serde_json::json;
+use std::sync::{mpsc, Arc};
+use std::time::Duration;
+use tokio::time::sleep;
+
+struct ComplexData {
+    /// There should be many more fields below in real life,
+    /// probaly wrapped in Arc<Mutex<T>> if you're writing a more serious client.
+    data: String,
+}
+
+struct TransmitterClient {
+    receiver: mpsc::Receiver<String>,
+    complex: ComplexData,
+    client: SocketIOClient,
+}
+
+impl TransmitterClient {
+    async fn connect(url: &str) -> Result<Self, SocketIOError> {
+        let (sender, receiver) = mpsc::channel::<String>();
+
+        let client = SocketIOClientBuilder::new(url)
+            .namespace("/admin")
+            .on("test", |payload: Payload, socket: SocketIOClient| {
+                async move {
+                    match payload {
+                        Payload::Text(values) => {
+                            if let Some(value) = values.first() {
+                                if value.is_string() {
+                                    socket
+                                        .try_transmitter::<mpsc::Sender<String>>()
+                                        .map_or_else(
+                                            |err| eprintln!("{}", err),
+                                            |tx| {
+                                                tx.send(String::from(value.as_str().unwrap()))
+                                                    .map_or_else(
+                                                        |err| eprintln!("{}", err),
+                                                        |_| {
+                                                            println!(
+                                                                "Data transmitted successfully"
+                                                            )
+                                                        },
+                                                    );
+                                            },
+                                        );
+                                }
+                            }
+                        }
```

Comment:
I am not a huge fan of the nesting tbh :D Could we either (or both):
1. Seperate this out into it's own async function?
2. Since we're only interested in `Payload::Text`, maybe we can shrink this further by having `if let Payload::Text(values) = payload {`